docs/writing-tests/test-runner.md

```diff
@@ -0,0 +1,301 @@
+
+- Use it locally and Chromatic on your CI.
+- Use Chromatic for visual and interaction tests and run other custom tests using the test runner.
+
+## Advanced configuration
+
+### Test hook API (experimental)
+
+The test-runner renders a story and executes its [play function](writing-stories/play-function.md) if one exists. However, certain behaviors are impossible to achieve via the play function, which executes in the browser. For example, if you want the test-runner to take visual snapshots for you, this is possible via Playwright/Jest but must be executed in Node.
+
+The test-runner exports test hooks that can be overridden globally to enable use cases like visual or DOM snapshots. These hooks give you access to the test lifecycle before and after the story is rendered.
+Listed below are the available hooks and an overview of how to use them.
+
+| Hook         | Description                                                                   |
+| ------------ | ----------------------------------------------------------------------------- |
+| `setup`      | Executes once before all the tests run<br/>`setup() {}`                       |
+| `preRender`  | Executes before a story is rendered<br/>`async preRender(page, context) {}`   |
+| `postRender` | Executes after the story is rendered<br/>`async postRender(page, context) {}` |
+
+<div class="aside">
+
+💡 These test hooks are experimental and may be subject to breaking changes. We encourage you to test as much as possible within the story's [play function](../writing-stories/play-function.md).
+
+</div>
+
+To enable the hooks API, you'll need to add a new configuration file inside your Storybook directory and set them up as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-hooks-example.js.mdx',
+    'common/storybook-test-runner-hooks-example.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+
+💡 Except for the `setup` function, all other functions run asynchronously. Both `preRender` and `postRender` functions include two additional arguments, a [Playwright page](https://playwright.dev/docs/pages) and a context object which contains the `id`, `title`, and the `name` of the story.
+
+</div>
+
+When the test-runner executes, your existing tests will go through the following lifecycle:
+
+- The `setup` function is executed before all the tests run.
+- The context object is generated containing the required information.
+- Playwright navigates to the story's page.
+- The `preRender` function is executed.
+- The story is rendered, and any existing `play` functions are executed.
+- The `postRender` function is executed.
+
+### Stories.json mode
+
+The test-runner transforms your story files into tests when testing a local Storybook. For a remote Storybook, it uses the Storybook's [stories.json](../configure/overview.md#feature-flags) file (a static index of all the stories) to run the tests.
```

Comment:
Is "remote Storybook" enough for users to understand what that means? Or might we need a more beginner friendly version of this?

Reply:
Also, we made a change that detects remote URLs and automatically runs with stories.json mode